File: Java/ExceptionHandling.java
//try....catch
//throw
//finally

//types of exceptions:-
//1)Arithmetic Exception
//2)ArrayOutOfBound Exception
//3)FileNotFound Exception


import java.io.*;
import java.lang.*;
import java.util.*;

class ExceptionHandling {
    public static void main(String[] args) {
        try {
            Scanner sc = new Scanner(System.in);
            System.out.println("Enter first value:");
            int x = sc.nextInt();
            System.out.println("Enter second value:");
            int y = sc.nextInt();
            if(y == 0) {
                throw new ArithmeticException ("Division by 0 is not possible");
            }
            int quotient = x / y;
            System.out.println("Answer is:"+quotient);
        }
        catch (Exception e) {
            System.out.println("An exception ocurred "+e);
        }
        finally {
            System.out.println("Finally block has been executed");
        }
    }
}

File: Python/ExceptionHandling.py
#example of exception
a = 12
b = 0    #in this case exception is executed
#b = 6    in this case no exception occured so after answer the finally block gets executed
try:
    print(a/b)
except Exception as e:
    print("An exception has occurred. Please check the code")
finally:
    print("Your code is executed.Thankyou.")




#example 2
try:
     x = int(input("Enter first number:"))
     y = int(input("Enter second number:"))
     ans = x/y
     print(ans)
except ZeroDivisionError:
     print("Zero Division Error has occurred.")
except Exception as e:
     print("An error occurred.", e)
finally:
     print("Your code is executed. Bye")



#example to create your own exception
class Error(Exception):
    pass
class VoteException(Error):          #new exception created
    pass

age = int(input("Enter your age as of 2023: "))
try:
    if(age>=18):
        print("You are eligible to vote")
    else:
        raise VoteException
except VoteException:
    print("This is an exception as you are not eligible to vote")